source3/libsmb/libsmb_setget.c

```diff
@@ -98,6 +98,19 @@ smbc_setDebug(SMBCCTX *c, int debug)
 	TALLOC_FREE(frame);
 }
 
+/** set configuration file */
+void smbc_setConfiguration(SMBCCTX *c, const char* file)
+{
+	if (lp_load_client(file))
```

Comment:
This will see the new "file" configuration appended / concatenated to the existing (smb.conf) config, right? In such a case the name "setConfiguration" might be a little misleading.

Reply:
No, It will be overwriting the configuration with that of defined in new configuration 'file'. The configuration not defined in 'file' would stay unchanged.
With this any code using libsmbclient APIs can maintain its own copy of smb.conf instead of relying on global smb.conf (from /etc/samba/smb.conf or $HOME environment variable)

Reply:
If you have the following in smb.conf:
```
[global]
  workgroup = SMBCONF
  log level = 10
```
and in custom.conf:
```
[global]
  workgroup = CUSTOM
```
then call
```
ctx = smbc_new_context();
smbc_init_context(ctx);
smbc_setConfiguration(ctx, "custom.conf");
```
"log level = 10" will still be configured following the smbc_setConfiguration() call, right?

Reply:
Yes